server.go

```diff
@@ -651,14 +665,29 @@ func (s *Server) HandleTracePacket(packet []byte) {
 		log.WithError(err).Warn("ParseSSF")
 		return
 	}
-	s.handleSSF(span, []string{"ssf_format:packet"})
+	// we want to keep track of this, because it's a client problem, but still
+	// handle the span normally
+	if span.Id == 0 {
+		reason := "reason:" + "empty_id"
+		s.Statsd.Count("ssf.error_total", 1, []string{"ssf_format:packet", "packet_type:ssf_metric", reason}, 1.0)
+		log.WithError(err).Warn("ParseSSF")
```

Comment:
We definitely should, because the span may contain metrics which we'd otherwise throw away.
If you're worried about skewing the metrics, we can sample if the final digit is 1 instead of 0 - I'll change that.

Reply:
It's not equivalent to returning early, because the span (and its constituent metrics) will still be sent on `spanChan` to be processed by the span workers. The only thing that won't be emitted are the two extra metrics we generate for 10% of spans: `veneur.ssf.spans.received_total` and `veneur.ssf.spans.tags_per_span`.

Reply:
OH. derp derp. my bad, i totally misread `handleSSF()`.